arch/arm64/include/irq.h

```diff
@@ -397,6 +397,15 @@ static inline void up_irq_restore(irqstate_t flags)
 #  define up_cpu_index() (0)
 #endif
 
+/****************************************************************************
+ * Schedule acceleration macros
+ ****************************************************************************/
+
+#define up_current_regs()      (this_task()->xcp.regs)
+#define up_this_task()         ((struct tcb_s *)(read_sysreg(tpidr_el1) & ~1ul))
```

Comment:
What is the meaning of this magical lsbit check ?

Reply:
whether we are running inside irq handler.

Reply:
Ok, I suggest giving a descriptive name to the magic number